.buildkite/pypi-publish-pipeline.yml

```diff
@@ -0,0 +1,114 @@
+## .buildkite/pypi-publish-pipeline.yml
+# Manually-triggered pipeline to build and publish Python packages to PyPI
+notify:
+  - if: 'build.branch =~ /^((main)|([0-9]+\.[0-9]+))$/ && (build.state == "failed" || pipeline.started_passing)'
+    slack:
+      channels:
+        - "#search-et-alerts"
+      message: "${BUILDKITE_MESSAGE}"
+
+# add paths for all the packages we want to build here
+# then make sure they're added to the matrix.setup.package_path lists in the steps below
+env:
+  SERVICE_PACKAGE_PATH: "app/connectors"
+  SDK_PACKAGE_PATH: "lib/connectors_sdk"
+
+steps:
+  - group: ":building_construction: Build Python Packages"
+    key: "build_package"
+    steps:
+      - label: ":python: Build Python {{matrix.python_version}} {{matrix.package_path}} package"
+        key: build_python_package
+        agents:
+          provider: "gcp"
+          machineType: "n1-standard-8"
+          useVault: true
+          image: family/enterprise-search-ubuntu-2204-connectors-py
+        matrix:
+          setup:
+            python_version:
+              - "3.10"
+              - "3.11"
+            package_path:
+              - "${SERVICE_PACKAGE_PATH}"
+              - "${SDK_PACKAGE_PATH}"
+        commands:
+          - "cd {{matrix.package_path}}"
+          - "python{{matrix.python_version}} -m pip install --upgrade build twine"
+          - "python{{matrix.python_version}} -m build"
+          - "ls -lah dist/"
+          - "python{{matrix.python_version}} -m twine check dist/*"
+        artifact_paths:
+          - "{{matrix.package_path}}/dist/*"
```

Comment:
I suggest moving these things to the Makefile. While the regular experience should be to trigger releases with buildkite, I think we'll want to be able to manually do it from our dev envs, or to easily be able to troubleshoot/dry-run locally. Keeping these types of "scripts" in a place where they can be reused will make that cleaner.